.circleci/config-2_1.yml

```diff
@@ -1164,6 +1164,18 @@ j11_separate_jobs: &j11_separate_jobs
         requires:
           - start_j11_jvm_dtests_vnode_repeat
           - j11_build
+    - start_j11_simulator_dtests:
```

Comment:
> But don't we need this also added in the pre-commit workflow, same as all other test suites?

:sigh: this build is getting too complicated... ill add

> we can skip j8 build with j11 runtime and directly go for j11 build with j17 

for now, since we support 8/11 ill keep the build doing 8/11; when we drop 8 and add 17 we can switch

Reply:
I wrote a script to confirm the change and found that if I manually applied on my own I would have messed up!

Here is the full diff of the workflows

```
Diff for ('java8_separate_tests', 'java11_separate_tests') is
left:
- start_java_dtest_jars_build
- java_dtest_jars_build
- start_jvm_upgrade_dtests
- java_jvm_upgrade_dtests
- start_jvm_upgrade_dtests_repeat
- java_jvm_upgrade_dtests_repeat
- start_upgrade_dtests
- java_upgrade_dtests
- start_java_cqlsh_tests_offheap
- start_java_upgrade_dtests_repeat
- java_upgrade_dtests_repeat
mismatch:
  java_cqlsh_dtests_py38_offheap:
    left:
    - start_java_cqlsh_tests_offheap
    - java_build
    right:
    - start_java_cqlsh-dtests-offheap
    - java_build
  java_cqlsh_dtests_py3_offheap:
    left:
    - start_java_cqlsh_tests_offheap
    - java_build
    right:
    - start_java_cqlsh-dtests-offheap
    - java_build
right:
- start_java_cqlsh-dtests-offheap

Diff for ('java8_pre-commit_tests', 'java11_pre-commit_tests') is
left:
- start_jvm_upgrade_dtests
- java_dtest_jars_build
- java_jvm_upgrade_dtests
- java_jvm_upgrade_dtests_repeat
- start_upgrade_dtests
- java_upgrade_dtests
- java_upgrade_dtests_repeat
- start_java_cqlsh_dtests_offheap
mismatch:
  java_dtests_offheap_repeat:
    left:
    - start_java_dtests_offheap
    - java_build
    right:
    - start_java_dtests_offheap_repeat
    - java_build
right:
- start_java_dtests_offheap_repeat

Diff for ('java8_separate_tests', 'java8_pre-commit_tests') is
left:
- j8_repeated_ant_test
- j11_repeated_ant_test

Diff for ('java11_separate_tests', 'java11_pre-commit_tests') is
left:
- j11_repeated_ant_test
```

Excluding differences not caused by this patch

```
$ circleci-verify.py --exclude upgrade dtest_jars cqlsh repeated_ant dtests_offheap
$
```

Here is the diff without the latest patch

```
$ circleci-verify.py --exclude upgrade dtest_jars cqlsh repeated_ant dtests_offheap
Diff for ('java8_pre-commit_tests', 'java11_pre-commit_tests') is
left:
- java_simulator_dtests
- java_simulator_dtests_repeat

Diff for ('java11_separate_tests', 'java11_pre-commit_tests') is
left:
- j11_simulator_dtests
- j11_simulator_dtests_repeat

```

Reply:
The issue you caught is naming which is known (tests vs dtests and _ vs -) and planned to be fixed across the board at once in a separate ticket. The only thing I noticed is that one of the repeated tests jobs will always be started in pre-commit if tests to run in a loop are added for that suite which can be made to match the rest, it is inconsistency but not some huge bug considering how many legit bugs(and I do not mean just missing tests) I found and how many tests I added.

`:sigh: this build is getting too complicated... ill add`
Well, we finally have all possible tests in CI and thanks to that we should be able to release. Also, now when it is finally clear that no one anymore uses the HIGHRES as is, we can switch to two configs - paid and free. I am looking forward to those changes. Let's see when people will have the time.

Reply:
[CASSANDRA-18007](https://issues.apache.org/jira/browse/CASSANDRA-18007)
CircleCI improvements and parity - this is an epic to address your concerns, if it doesn't feel free to add whatever is missing to the epic. Thanks